Solution1/BusinessLayer/Services/WeatherApiService.cs

```diff
@@ -15,11 +20,30 @@ public WeatherApiService(HttpClient httpClient)
             _httpClient=httpClient;
         }
 
-        public Task<WeatherApiDTO> GetByCityNameAsync(string cityName)
+        public async Task<WeatherApiDTO> GetByCityNameAsync(string cityName)
```

Comment:
why you added async and await here.....?

Reply:
+